File: test/test-question-do-plus-loop.js
const gb = require("./gbtest")(__filename);

test("?do...+loop", () => {
  gb.run();
  const result1 = [0, 3, 6, 9];
  const result2 = [0, gb.int(-1)];
  expect(gb.stack).toEqual([...result1, ...result2]);
});

File: test/test-question-do-plus-loop.fs
: test-1
  10 0 ?do i 3 +loop ;

: test-2
  -1 0 ?do  i -1 +loop ;

: test-3
  0 0 ?do  i  -1 +loop ;

: main
  test-1
  test-2
  test-3
;
